File: src/TelegramBot/Domain/Records/TransportStop.cs
namespace WhereIsTheBus.TelegramBot.Domain.Records;

internal sealed record TransportStop(int Id, string Name, StrictDirection Direction, int TimeToArrive)
{
    public bool Equals(TransportStop? other)
    {
        if (ReferenceEquals(null, other)) return false;
        if (ReferenceEquals(this, other)) return true;
        return Id == other.Id;
    }

    public override int GetHashCode()
    {
        return Id;
    }
}

File: src/IzhTransport/IzhTransport/Models/TransportStop.swift
//
//  TransportStop.swift
//  IzhTransport
//
//  Created by Nikita Karalius on 22.05.2022.
//

import Foundation

struct TransportStop {
    let id: Int
    let name: String
    let direction: StrictDirection
    let timeToArrive: TimeToArrive
}
